mapviz/CMakeLists.txt

```diff
@@ -122,6 +122,19 @@ add_library(rqt_${PROJECT_NAME}
   ${SRC_FILES}
   ${RCC_SRCS}
 )
+target_include_directories(rqt_${PROJECT_NAME} 
+  PUBLIC
+    include
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(rqt_${PROJECT_NAME} SYSTEM
+  PUBLIC
+    ${Qt_INCLUDE_DIRS}
+    ${OpenGL_INCLUDE_DIR}
+    ${GLEW_INCLUDE_DIRS}
+    ${OpenCV_INCLUDE_DIRS}
+    ${YamlCpp_INCLUDE_DIR})
```

Comment:
We can probably clean up the include directories a little more; something to note is that if you use the modern syntax for specifying your `target_link_libraries` (as we do with `OpenGL::GL`, `Qt5::Concurrent`, etc. below), it will automatically add the header directories to the include path without needing to manually specify them.  So, we should at least be able to remove Qt and OpenGL from this list, and I believe there are also appropriate targets we can use for including OpenCV (https://stackoverflow.com/questions/55640570/using-opencvs-cmake-module-targets-instead-of-including-all-libraries-directly).  There might also be similar things we have use for GLEW or YamlCpp, I haven't looked them up.

Reply:
I have converted OpenGL_INCLUDE_DIR, GLEW_INCLUDE_DIRS, and OpenCV_INCLUDE_DIRS to their target equivalents. However, I'm not finding a target equivalent for YamlCpp. The generated targets cmake file appears to be in a CMake version that is too old for Modern CMake (<3.0)